crates/python/pyproject.toml

```diff
@@ -44,6 +44,7 @@ sdist-include = ["README.md"]
 
 [tool.poetry.dependencies]
 python = "^3.8"
+opentelemetry-api = "^1.20.0"
```

Comment:
After becoming more familiar with the [pyo3_opentelemetry docs](https://docs.rs/pyo3-opentelemetry/latest/pyo3_opentelemetry/), it looks like it is required for Python consumers.

Reply:
Good point. Yes it would definitely be required, but only when using the tracing subscriber. I'm thinking we should treat that as an extra, so we don't burden anyone with unwanted dependencies.

I suppose that means including it as an [optional-dependency](https://peps.python.org/pep-0631/#optional-dependencies) and that I should define `tool.poetry.extras.tracing-opentelemetry`.

Reply:
That looks right to me! Since this is the first extra, when the releases as an RC, it would be good to validate that a fresh project doesn't run into any missing dependency errors, that way we're confident the full release won't break anyone.